File: Map/Map/Map/statement/AssignStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Map.domain;
using Map.expression;

namespace Map.statement
{
    class AssignStatement : IStatement
    {
        private String varName;
        private IExpression expr;

        public AssignStatement(String name,
                               IExpression ex)
        {
            varName = name;
            expr = ex;
        }

        public ProgramState execute(ProgramState ps)
        {
            IDict<String, int> symb = ps.symbolTable;
            int res = expr.eval(symb);
            if (symb.contains(varName))
                symb.update(varName, res);
            else
                symb.add(varName, res);
            return null;

        }

        public override string ToString()
        {
            return varName + "=" + expr;
        }
    }
}

File: Map/Z_InterpretorWLock/src/statement/AssignStatement.java
package statement;

import domain.IDictionary;
import domain.IHeap;
import domain.PrgState;
import expression.IExpression;

public class AssignStatement implements IStatement
{
	private String varName;
	private IExpression expr;
	
	public AssignStatement(String vn, IExpression e)
	{
		varName = vn;
		expr = e;
	}

	@Override
	public PrgState execute(PrgState ps) 
	{
		IDictionary<String,Integer> symbTab = ps.getSymbolTable();
		IHeap<Integer,Integer> heap = ps.getHeap();
		int result = expr.eval(symbTab,heap);
		if(symbTab.contains(varName))
			symbTab.update(varName, result);
		else
			symbTab.add(varName, result);     //!
		return null;
	}
	
	@Override
	public String toString()
	{
		return varName+"="+expr;
	}
}
